### DISCUSSION TITLE
Add limit Download per torrent (not globally)

### DISCUSSION THREAD

**Discussion by ELOUFIR** [KiraLT/torrent-stream-server#273] [Ideas]
Hi everyone
I am very happy with this incredible project, everything I wanted I found it already integrated.

I just have one question, why don't we have control of the torrent download limit? I mean for each torrent not globally !!

Thanks

**Comment by KiraLT** [comment]
There is a reason behind this logic. In most cases, you wish to deploy this project to a server, so you could access using any device (PC, mobile, or even TV). The server has a global resource limit, on which it works without issues- in most cases the bottleneck is CPU. The faster you download, the more CPU is required to process all downloads in real-time. So you set a global limit and in all cases, the server runs smoothly without crashing. If you running only one torrent, it will use full speed, if you running more, they will share the same speed. So if you add a lot of torrents, they will just slow down without reaching the limit and crashing. 

You can increase the global download limit in the [config](https://github.com/KiraLT/torrent-stream-server/wiki/Configuration) if you running on a more powerful machine.

So that was the reason for this implementation, but maybe your use case is different. Could you tell why you need limitations/torrent?

**Reply by ELOUFIR** [reply]
Thank you for your quick reply

My concern is that if I have more than one torrent and one of them has more seeders than the others, I think it will consume more speed (dominate).
And for streaming, I think 1~2mb/s is enough as a limit for each torrent.

My Best regards

**Reply by KiraLT** [reply]
In theory, one torrent shouldn't consume all the speed and block others - they should share equally.

Well, maybe it would be useful to combine global speed with / torrent speed. For example, 10Mb/s global speed + 2Mb/s / torrent speed. However, the library we use ([webtorrent](https://github.com/webtorrent/webtorrent)) supports only global limits, so at the moment we can't have this functionality.
